wallet/handler.go

```diff
@@ -259,6 +266,75 @@ func DescribeKey(store Store, req *DescribeKeyRequest) (*DescribeKeyResponse, er
 	return resp, nil
 }
 
+type RotateKeyRequest struct {
+	Wallet            string
+	Passphrase        string
+	NewPublicKey      string
+	TXBlockHeight     uint32
+	TargetBlockHeight uint32
+}
+
+type RotateKeyResponse struct {
+	MasterPublicKey   string `json:"master_public_key"`
+	NewPublicKey      string `json:"new_public_key"`
+	Base64Transaction string `json:"base64_transaction"`
+}
+
+func RotateKey(store Store, req *RotateKeyRequest) (*RotateKeyResponse, error) {
+	w, err := getWallet(store, req.Wallet, req.Passphrase)
+	if err != nil {
+		return nil, err
+	}
+
+	mKeyPair, err := w.GetMasterKeyPair()
+	if err != nil {
+		return nil, err
+	}
+
+	newKeyPair, err := w.DescribeKeyPair(req.NewPublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get new key pair: %w", err)
+	}
+
```

Comment:
We should check here whether the key is tainted and fail. We do not want to let people rotate to a tainted key.

Reply:
Yes we should. Key rotation is a security "mechanism", we should be fool proof, and help the user in avoiding using unsecure key.